pkg/storage/tsdb/caching_bucket.go

```diff
@@ -0,0 +1,73 @@
+package tsdb
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/thanos-io/thanos/pkg/cache"
+	"github.com/thanos-io/thanos/pkg/cacheutil"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	storecache "github.com/thanos-io/thanos/pkg/store/cache"
+)
+
+type CachingBucketConfig struct {
+	Backend   string                    `yaml:"backend"`
+	Memcached MemcachedIndexCacheConfig `yaml:"memcached"`
+
+	CachingConfig storecache.CachingBucketConfig `yaml:"caching"`
+}
+
+func (cfg *CachingBucketConfig) RegisterFlags(f *flag.FlagSet) {
+	cfg.RegisterFlagsWithPrefix(f, "experimental.tsdb.bucket-store.index-cache.")
```

Comment:
Is `index-cache` intended here? Do we need this `RegisterFlags()` function at all?